### DISCUSSION TITLE
KMemo: adding the string ")" after the paragraph letter

### DISCUSSION THREAD

**Discussion by radekjindra** [kryslt/KControls#60] [Q&A]
Hello, I would like to ask for advice. I need to add the line ")" after the paragraph letter. a) b) c)

I have for now

TB:=Protokol_ZKP.Blocks.AddTextBlock('Non-metallic materials for line conductors are preferred over metallic ones with insulating contacts.');
TB.TextStyle.Font.Name:='Times New Roman';
TB.TextStyle.Font.Color:=clRed;
TB.TextStyle.Font.Style:=[fsBold];
TB.TextStyle.Font.Size:=10;
PA:=Protokol_ZKP.Blocks.AddParagraph;
PA.ParaStyle.NumberingListLevel := 0;
PA.Numbering := pnuLetterLo;
PA.ParaStyle.HAlign:=halJustify;
PA.NumberingListLevel.FirstIndent:=10;
PA.NumberingListLevel.LeftIndent :=30;
PA.ParaStyle.NumberStartAt := 1; 

If I save it in protokol.rtf, it doesn't appear in the file from the letter "a" but from "f" even though I used 

PA.ParaStyle.NumberStartAt := 1;

But on the screen the paragraph appears from "a".

Thank you

**Comment by kryslt** [comment]
There is support for custom numbering formats which work the same as RTF numbering formats. 
However, their usage in KMemo is not obvious. 
I will convert this to an issue and add some code that demonstrates their usage.

Please create another discussions or issues for your other questions.

**Comment by kryslt** [comment]
Use TKMemoListLevel.LastNumberSeparator as shown in #61 and the corresponding commit.
Eg.
`PA.NumberingListLevel.LastNumberSeparator := ')';`
Please note this will change the separator for any paragraphs using this list.
If you have eg. 'a. b. c.' in this list which is used for some paragraphs and need to use 'a) b) c)' for another paragraphs then you need to create another list for them. In that case use the modified method KMemo.ListTable.ListByNumbering where you can input the desired separator.
